### ISSUE TITLE
False positives for Layout/SpaceAroundOperators

### ISSUE THREAD

**Issue by avdi** [rubocop/rubocop#13586]
This seems to be a regression in 1.69.2

--------

## Expected behavior

With the default (or explicitly set) option of `AllowForAlignment: true`, this code should pass:

```ruby
      def last_sync_is_stale?
        @wants_resync                                      ||
          @connection.uid_validity       != @uidvalidity   ||
          @connection.last_synced_exists != @server_exists ||
          @connection.highest_synced_uid != highest_uid    ||
          @connection.last_mailbox_sync  <= full_sync_timespan.ago(clock.call)
      end
```

## Actual behavior

The last line of logic (only) is flagged:

```
scratch.rb:6:36: C: [Correctable] Layout/SpaceAroundOperators: Operator <= should be surrounded by a single space.
    @connection.last_mailbox_sync  <= full_sync_timespan.ago(clock.call)
```

## Steps to reproduce the problem

1. Copy the code above into scratch.rb
2. Run `rubocop scratch.rb`

## RuboCop version

```
1.69.2 (using Parser 3.3.6.0, rubocop-ast 1.37.0, analyzing as Ruby 3.2, running on ruby 3.2.5) [x86_64-linux]
  - rubocop-capybara 2.21.0
  - rubocop-factory_bot 2.26.1
  - rubocop-performance 1.23.0
  - rubocop-rails 2.27.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 3.3.0
  - rubocop-rspec_rails 2.30.0
 ```
